### PR TITLE
Add make check and Windows support

### PR DESCRIPTION
**Pull request by ryanvolz**
I've had my own log4cpp recipe for a little while. Now that conda-forge has one, it would be good to merge the couple features I have working (make check, Windows) so I can drop my recipe.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,24 +9,31 @@ source:
   fn: {{ name }}-{{ version }}.tgz
   url: https://sourceforge.net/projects/{{ name }}/files/{{ name }}-1.1.x%20%28new%29/{{ name }}-1.1/{{ name }}-{{ version }}.tar.gz
   sha256: 2cbbea55a5d6895c9f0116a9a9ce3afb86df383cd05c9d6c1a4238e5e5c8f51d
+  patches:
+    - osx_remove_manual_test.patch  # [osx]
 
 build:
-  skip: true  # [win]
-  number: 0
+  number: 2
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: please, change build number to 1

**Comment by marcelotrevisani** [comment]
After you correct the build number to 1 and it passed in our CI, I will merge that

Thanks for your help!
:)

File: recipe/bld.bat [hunk 2]
```diff
@@ -0,0 +1,21 @@
+setlocal EnableDelayedExpansion
+
+:: Make a build folder and change to it
+mkdir build
+cd build
+
+:: configure
+cmake -G "NMake Makefiles" ^
```
Comment (marcelotrevisani) [inline recipe/bld.bat]: Instead of 
```
-G "NMake Makefiles"
```
could you please try 
```
-G "Ninja"
```
and just do a
```
ninja
ninja install
```

**Comment by ryanvolz** [comment]
Requested changes done!

**Comment by marcelotrevisani** [comment]
Looks good to me. After the CI passed for windows I will merge that

Thanks @ryanvolz !
:)

File: recipe/meta.yaml [hunk 3]
```diff
@@ -9,24 +9,31 @@ source:
...
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - make
-    - autoconf
+    - make  # [unix]
+    - autoconf  # [unix]
+    - cmake  # [win]
```
Comment (marcelotrevisani) [inline recipe/meta.yaml:24]: Oh, I forgot to mention to add ``ninja`` as a dependency. Please, add
```
- ninja  # [win]
```

File: recipe/bld.bat [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+:: configure
+cmake -G "Ninja" ^
+      -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+      -DCMAKE_PREFIX_PATH:PATH="%LIBRARY_PREFIX%" ^
+      -DCMAKE_BUILD_TYPE:STRING=Release ^
+      ..
+if errorlevel 1 exit 1
+
+:: build
+ninja
```
Comment (marcelotrevisani) [inline recipe/bld.bat]: we can let this just as 
```
ninja install
```
it will compile and install it

**Comment by marcelotrevisani** [comment]
Thanks @ryanvolz !

**Pull request merged**